internal/engine/upper.go

```diff
@@ -109,20 +109,22 @@ func (u Upper) CreateManifests(ctx context.Context, manifests []model.Manifest,
 		Manifests:   manifests,
 	})
 
+	hudErrorCh := make(chan error)
+
 	// Run the HUD in the background
 	go func() {
-		err := u.hud.Run(ctx, u.store)
-		if err != nil {
-			//TODO(matt) this might not be the best thing to do with an error - seems easy to miss
-			logger.Get(ctx).Infof("error in hud: %v", err)
-		}
+		hudErrorCh <- u.hud.Run(ctx, u.store)
+		close(hudErrorCh)
 	}()
 
 	for {
 		timer := u.timerMaker(refreshInterval)
 
 		select {
 		case <-ctx.Done():
+			// make sure the hud has had a chance to clean up
+			<-hudErrorCh
```

Comment:
do we need to allow the HUD time to clean up if we return from other place sin this func as well (eg ln 153)?

Reply:
yep, good catch!